projects/hslayers/src/common/add-to-map/add-to-map.component.ts

```diff
@@ -0,0 +1,60 @@
+import {CommonModule} from '@angular/common';
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+import {HsLanguageModule} from '../../components/language/language.module';
+
+@Component({
+  selector: 'hs-add-to-map',
+  standalone: true,
+  template: `
+    <button
+      class="btn btn-primary w-100"
+      [disabled]="disabled"
+      [title]="title | translateHs"
+      (click)="addLayer()"
+      [ngClass]="classes"
+    >
+      <span
+        *ngIf="!loading; else loader"
+        class="d-flex justify-content-center align-items-baseline gap-1"
+      >
+        <i class="icon-plus"> </i>{{ 'COMMON.addToMap' | translateHs }}
+      </span>
+      <ng-template #loader>
+        <span>
+          <span class="hs-loader"></span>&emsp;
+          {{ 'COMMON.uploading' | translateHs }}
+        </span>
+      </ng-template>
+      <ng-content></ng-content>
+    </button>
+  `,
+  styles: [
+    `
+      :host {
+        display: block;
+      }
+    `,
+  ],
+  imports: [CommonModule, HsLanguageModule],
+})
+/**
+ * title: Translation string to be used title
+ */
+// eslint-disable-next-line @angular-eslint/component-class-suffix
+export class HsAddToMapButton {
```

Comment:
Any reason why not to call this class 
```suggestion
export class HsAddToMapButtonComponent {
```
?

Reply:
[2fbf277](https://github.com/hslayers/hslayers-ng/pull/4468/commits/2fbf27703cb93a15bb3604b2d187c615a9374370)